File: Python/01_Strings_Arrays/test_is_one_edit.py
import unittest
from IsOneEdit import is_one_edit

class TestIsOneEdit(unittest.TestCase):

    def test_regular(self):
        self.assertTrue(is_one_edit("a", "ab")) #true
        self.assertFalse(is_one_edit("a", "abc")) #false
        self.assertTrue(is_one_edit("ac", "ab")) #true
        self.assertFalse(is_one_edit("acee", "ab")) #false
        self.assertTrue(is_one_edit("ab", "abc")) #true
        self.assertFalse(is_one_edit("aee", "aecc")) #false

if __name__ == '__main__':
    unittest.main()

File: Java/CTCI/src/strings_and_arrays/TestIsOneEdit.java
package strings_and_arrays;

import static org.junit.Assert.*;
import org.junit.Test;

public class TestIsOneEdit {
	@Test
    public void testIsOneEdit() {
               
        assertTrue(IsOneEdit.is_one_edit("abc", "abcd")); // true
        assertTrue(IsOneEdit.is_one_edit("abc", "ab")); //true
        assertTrue(IsOneEdit.is_one_edit("abc", "abb")); // true
        assertTrue(IsOneEdit.is_one_edit("abc", "bbc")); //true
        assertFalse(IsOneEdit.is_one_edit("abc", "abdd")); //false
        assertFalse(IsOneEdit.is_one_edit("abc", "a")); //false
        assertFalse(IsOneEdit.is_one_edit("abc", "add")); //false
        assertFalse(IsOneEdit.is_one_edit("aaaaa", "bcd")); //false
    }
}
